csrc/scheduler/reduction.cpp

```diff
@@ -1225,6 +1226,12 @@ void scheduleReduction(Fusion* fusion, const ReductionParams& rparams) {
       cached_outputs);
 
   scheduler_utils::promoteProducerMemoryTypes(fusion, cached_inputs);
+
+  // TODO(#1401): We could let segmentation split a partially alias-producing
+  // fusion into an alias-only segment and the rest. This way, the rest of the
+  // fusion (which has fewer expressions) can potentially find a better
+  // scheduler and we need to call markAliases only in NoOpScheduler.
+  markAliases(fusion);
```

Comment:
Am I understanding this correctly that `markAliases` should potentially be called in every scheduler? should we have added it in maybe transpose scheduler as well?

Reply:
Yes! I'm adding it incrementally to reduce the blast radius. I was also hoping (as the comment says) segmentation is improved so only NoOpScheduler needs to mark aliases, but I haven't prioritized that yet without a concrete need.

Reply:
Got'ya.